### ISSUE TITLE
FATAL ERROR: Not enough free space in your /mnt/ system!

### ISSUE THREAD

**Issue by alex9434** [FriendlyNeighborhoodShane/MinMicroG#72]
Trying to install MinMicroG on Android 14 (crDroid 10) on Poco X6 5G (garmet). I get the following error message:
`!!! FATAL ERROR: Not enough free space in your /mnt/ system!`
I get this message using TWRP or Orange Fox to flash MinMicroG and with all variants.
How can I solve this?

**Comment by Izaic** [comment]
Same issue

**Comment by BlueStarOne** [comment]
After 3 days I finally found how to fix this.

In facts you need to flash the zip as a magisk addon. Minmicrog will then be installed with root permissions, as of you had flashed it in the recovery. 

It worked fine for me, after I tried many custom recoveries and many thing to fix this error.

P.S : Extremely late answer, this message is more for other users (I guess you already solved it)

**Comment by kiecla** [comment]
Seeing the same issue on Lineage Recovery
edit: Duplicate of https://github.com/FriendlyNeighborhoodShane/MinMicroG/issues/60

**Comment by alex9434** [comment]
@BlueStarOne you are right, that is a work-around. However, that requires to root with Magisk. On previous Android versions I could use TWRP to directly add MinMicroG to the rom without rooting. This means that the rom would pass SafetyNet without further modifications and enables to use banking apps etc. Having to install Magisk makes this much harder (need to install and constantly update additional modules) and in some cases impossible because of root detection.

For me it is really desirable to install with recovery (TWRP) instead of rooting.

**Comment by BlueStarOne** [comment]
> For me it is really desirable to install with recovery (TWRP) instead of rooting.

Of course, you're totally right, i'd also better have took this option, but sadly, after retrying 5 differents things to solve the original issue, my only option left was to use magisk (i would have installed it anyway).
Maybe i have to use the safety net plugin and shamiko+deny list to make some apps work, but at least microg is fully working for me

**Comment by Lanchon** [comment]
this is a problem with your android build and there is nothing that microG can do about it. either your phone has a small system partition and the system grew to the point there is very little space left, or -most likely- your phone is newer and has dynamic partitions, but your rom builder did not allocate any extra free space for flashing stuff like gapps in the dynamically sized system partition.

talk to your rom builder for a solution:

- on older, fixed partition devices: they can debloat their build so that gapps will fit. if gapps fit, microg will fit too.
- on dynamic partition device: they can increase the size of system to reserve free space.
  - LineageOS does this; see the WITH_GMS build flag (which actually cancels this reservation if defined).

what you can do:

- debloat your system partition manually to free up some space.
  - but you will have to do this after every update, and the addon.d mechanism to survive flashes will not work.
- install microg systemlessly, as via magisk or kernelsu.
  - meaning that microg looks as if installed in system, but is actually stored in the data partition where you have free space.
- repartition your device to give it a larger system partition.
  - but many newer devices are suspected of expecting a specific GPT (partition table) due to their secure boot implementation, and irrecoverably brick if the GPT is altered.
  - for more info on repartitioning check out the Lanchon REPIT project.
